### ISSUE TITLE
qwen3-omni gspo: vllm使用sever模式，rollout返回乱码（已检测swift infer正常）

### ISSUE THREAD

**Issue by wwxs123W** [modelscope/ms-swift#9114]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

qwen3-omni vllm使用sever模式，rollout返回乱码（已检测swift infer正常），返回完全是乱码，应该是权重随机初始化时的现象。

### How to Reproduce / 如何复现

1.rollout侧代码
`export NCCL_DEBUG=INFO
export NCCL_TIMEOUT=3600
export NCCL_ASYNC_ERROR_HANDLING=1
export NCCL_SOCKET_IFNAME=eth0
export NCCL_IB_DISABLE=1
export NCCL_P2P_DISABLE=1
export OMP_NUM_THREADS=8
export PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True

MAX_PIXELS=602112 \
ENABLE_AUDIO_OUTPUT=0 \
FPS_MAX_FRAMES=32 \
USE_AUDIO_IN_VIDEO=1 \
CUDA_VISIBLE_DEVICES=0,1,2,3 \
swift rollout \
    --model /path/to/Qwen3-Omni-30B-A3B-Instruct \
    --vllm_tensor_parallel_size 4 \
    --vllm_gpu_memory_utilization 0.85 \
    --vllm_max_num_seqs 8 \
    --external_plugins examples/train/grpo/plugin/plugin.py \
    --system examples/train/grpo/prompt.txt \
    --max_length 32768 \
    --enable_thinking false \
`
2.训练代码
`export MASTER_ADDR="127.0.0.1"
export MASTER_PORT=12345
export NNODES=1
export NODE_RANK=0

export NCCL_DEBUG=WARN
export NCCL_TIMEOUT=3600
export NCCL_ASYNC_ERROR_HANDLING=1
export PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True
export OMP_NUM_THREADS=8

export VLLM_WORKER_MULTIPROC_METHOD=spawn
export VLLM_CUSTOM_ALL_REDUCE=0

export NCCL_IB_DISABLE=1
export NCCL_SOCKET_IFNAME=eth0
export NCCL_P2P_DISABLE=1

GBS=16
NUM_GENERATIONS=8

export MAX_PIXELS=602112
export FPS_MAX_FRAMES=32
export FPS_MIN_FRAMES=4
export FPS=2.0
export IMAGE_MIN_TOKEN_NUM=576
export IMAGE_MAX_TOKEN_NUM=5832
export VIDEO_MIN_TOKEN_NUM=128
export VIDEO_MAX_TOKEN_NUM=768

export ENCODE_TIMEOUT=300
export USE_AUDIO_IN_VIDEO=1
export SWIFT_PATCH_CONV3D=1
export ENABLE_AUDIO_OUTPUT=0
export CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7

echo ">>> Starting single-node 8-GPU training, Master: $MASTER_ADDR:$MASTER_PORT"

NPROC_PER_NODE=8 \
NNODES=1 \
NODE_RANK=0 \
MASTER_ADDR=$MASTER_ADDR \
MASTER_PORT=$MASTER_PORT \
megatron rlhf \
    --rlhf_type grpo \
    --model /path/to/Qwen3-Omni-30B-A3B-Instruct \
    --tuner_type lora \
    --dataset /path/to/data \
    --reward_funcs external_r1v_acc format \
    --reward_weights 1 1.0 \
    --external_plugins ms-swift/examples/train/grpo/plugin/plugin.py \
    \
    --bf16 true \
    --num_train_epochs 1 \
    --micro_batch_size 1 \
    --tensor_model_parallel_size 4 \
    --expert_model_parallel_size 4 \
    --pipeline_model_parallel_size 1 \
    --context_parallel_size 1 \
    --sequence_parallel true \
    --attention_backend flash \
    --padding_free false \
    \
    --recompute_granularity selective \
    --offload_model false \
    --offload_optimizer true \
    --optimizer_cpu_offload true \
    --use_precision_aware_optimizer \
    --sleep_level 0 \
    \
    --moe_permute_fusion true \
    --moe_grouped_gemm true \
    --moe_shared_expert_overlap true \
    --moe_aux_loss_coeff 1e-3 \
    \
    --freeze_vit true \
    --freeze_aligner true \
    --freeze_parameters talker code2wav \
    --vit_gradient_checkpointing true \
    \
    --lr 1e-6 \
    --min_lr 1e-7 \
    --lr_warmup_fraction 0.05 \
    --beta 0.00 \
    --epsilon 3e-4 \
    --epsilon_high 4e-4 \
    --loss_type grpo \
    --importance_sampling_level sequence_token \
    --system examples/train/grpo/prompt.txt \
    \
    --save_total_limit 20 \
    --save_steps 200 \
    --eval_steps 50 \
    --logging_steps 5 \
    --no_save_optim true \
    --no_save_rng true \
    --output_dir path/to/outputs \
    \
    --global_batch_size $GBS \
    --num_generations $NUM_GENERATIONS \
    --steps_per_generation 1 \
    --max_completion_length 2048 \
    \
    --dataset_num_proc 1 \
    --dataloader_num_workers 0 \
    --temperature 1.0 \
    --top_p 0.99 \
    --top_k 50 \
    --log_completions true \
    --dynamic_sample false \
    --overlong_filter true \
    --truncation_strategy delete \
    --seed 42 \
    \
    --use_vllm true \
    --vllm_mode server \
    --vllm_server_host 10.245.251.49 \
    --vllm_server_port 8000 \
    --vllm_server_group_port 51216 \
    --vllm_server_timeout 300 \
    --max_length 32768
`

**Comment by hjh0119** [comment]
plz retry with main branch

**Comment by wwxs123W** [comment]
> plz retry with main branch

我已经尝试，问题已解决，感谢！

**Comment by ooochen-30** [comment]
@wwxs123W @hjh0119 麻烦问一下你用了哪个版本的ms_swift，我在qwen3.5-35b-a3b的grpo训练时遇到了一样的问题，尝试了vllm serve, swift deploy, transformer单测推理都是ok的，swift rollout就会乱码，更换了ms-swift 4.2.1,4.2.3都会乱码，最新版本又有其他的问题

**Comment by hjh0119** [comment]
@ooochen-30 plz open a new issue

**Comment by tanABCC** [comment]
@wwxs123W 可以share一下如何解决的吗？
